src/Features/VisualBasic/Portable/MakeMethodSynchronous/VisualBasicMakeMethodSynchronousCodeFixProvider.vb

```diff
@@ -0,0 +1,136 @@
+                ' Convert this to a 'Sub' method.
+                Dim subStatement = SyntaxFactory.SubStatement(
+                    functionStatement.AttributeLists,
+                    functionStatement.Modifiers,
+                    SyntaxFactory.Token(SyntaxKind.SubKeyword).WithTriviaFrom(functionStatement.SubOrFunctionKeyword),
+                    functionStatement.Identifier,
+                    functionStatement.TypeParameterList,
+                    functionStatement.ParameterList,
+                    functionStatement.AsClause,
+                    functionStatement.HandlesClause,
+                    functionStatement.ImplementsClause)
+
+                subStatement = subStatement.RemoveNode(subStatement.AsClause, SyntaxRemoveOptions.KeepTrailingTrivia)
+                subStatement = RemoveAsyncKeyword(subStatement)
+
+                Dim endSubStatement = SyntaxFactory.EndSubStatement(
+                    node.EndSubOrFunctionStatement.EndKeyword,
+                    SyntaxFactory.Token(SyntaxKind.SubKeyword).WithTriviaFrom(node.EndSubOrFunctionStatement.BlockKeyword))
+
+                Return SyntaxFactory.SubBlock(subStatement, node.Statements, endSubStatement)
+            Else
+                Dim newFunctionStatement = RemoveAsyncKeyword(functionStatement)
+                Return node.WithSubOrFunctionStatement(newFunctionStatement)
+            End If
+        End Function
+
+        Private Function FixSubBlock(node As MethodBlockSyntax) As SyntaxNode
+            Dim newSubStatement = RemoveAsyncKeyword(node.SubOrFunctionStatement)
+            Return node.WithSubOrFunctionStatement(newSubStatement)
+        End Function
+
+        Private Shared Function RemoveAsyncKeyword(subOrFunctionStatement As MethodStatementSyntax) As MethodStatementSyntax
+            Dim modifiers = subOrFunctionStatement.Modifiers
+            Dim asyncTokenIndex = modifiers.IndexOf(SyntaxKind.AsyncKeyword)
+
+            Dim newSubKeyword = subOrFunctionStatement.SubOrFunctionKeyword
+            Dim newModifiers As SyntaxTokenList
+            If asyncTokenIndex = 0 Then
+                ' Have to move the trivia on teh async token appropriately.
+                Dim asyncLeadingTrivia = modifiers(0).LeadingTrivia
+
+                If modifiers.Count > 1 Then
+                    ' Move the trivia to the next modifier;
+                    newModifiers = modifiers.Replace(
+                        modifiers(1),
+                        modifiers(1).WithPrependedLeadingTrivia(asyncLeadingTrivia))
+                    newModifiers = newModifiers.RemoveAt(0)
+                Else
+                    ' move it to the return type.
+                    newModifiers = modifiers.RemoveAt(0)
+                    newSubKeyword = newSubKeyword.WithPrependedLeadingTrivia(asyncLeadingTrivia)
+                End If
+            Else
+                newModifiers = modifiers.RemoveAt(asyncTokenIndex)
```

Comment:
i'm really not concerned about this.  The same logic applies.  in literally the 99.999% case there will be no interesting trivia on this particular token.  And in the case where there is some trivia talking aobut this token, I believe we should remove it because... well... we're removing the token.